crates/interpreter/src/exec.rs

```diff
@@ -1032,8 +1038,9 @@ impl<'m> Thread<'m> {
             LabelKind::Block | LabelKind::If => type_.results.len(),
             LabelKind::Loop(_) => type_.params.len(),
         };
-        let label = Label { arity, kind, values };
+        let label = Label { arity, kind, values_cnt: values.len() };
         self.labels().push(label);
+        self.values().extend_from_slice(&*values);
```

Comment:
This is typically an example where we don't need to pop to push directly after (and the whole point of this effort). It should be possible to remove line 1036 and not add this line.